packages/financial-templates-lib/test/price-feed/CreatePriceFeed.js

```diff
@@ -713,6 +713,119 @@ contract("CreatePriceFeed.js", function(accounts) {
+
+    assert.isNull(expressionPriceFeed);
+  });
+
+  it("ExpressionPriceFeed: invalid config, symbol not found", async function() {
+    const config = {
+      type: "expression",
+      expression: "mysymbol * 2"
+    };
+
+    const expressionPriceFeed = await createPriceFeed(logger, web3, networker, getTime, config);
+
+    assert.isNull(expressionPriceFeed);
+  });
+
+  it("ExpressionPriceFeed: customFeeds", async function() {
+    const config = {
+      type: "expression",
+      expression: "mysymbol * 2",
+      customFeeds: {
+        mysymbol: {
+          type: "cryptowatch",
+          apiKey,
+          exchange,
+          pair,
+          lookback,
+          minTimeBetweenUpdates
+        }
+      }
+    };
+
+    const expressionPriceFeed = await createPriceFeed(logger, web3, networker, getTime, config);
+
+    assert.isNotNull(expressionPriceFeed);
+    assert.exists(expressionPriceFeed.priceFeedMap["mysymbol"]);
+  });
+
+  it("ExpressionPriceFeed: inherited config", async function() {
+    const config = {
+      type: "expression",
+      expression: "mysymbol * 2",
+      apiKey,
+      exchange,
+      pair,
+      lookback,
+      minTimeBetweenUpdates,
+      customFeeds: {
+        mysymbol: {
+          type: "cryptowatch"
+        }
+      }
+    };
+
+    const expressionPriceFeed = await createPriceFeed(logger, web3, networker, getTime, config);
+
+    assert.isNotNull(expressionPriceFeed);
+    assert.exists(expressionPriceFeed.priceFeedMap["mysymbol"]);
+    assert.equal(expressionPriceFeed.priceFeedMap["mysymbol"].lookback, lookback);
+  });
+
+  it("ExpressionPriceFeed: invalid config, no expression", async function() {
+    const config = {
+      type: "expression"
+    };
+
+    const expressionPriceFeed = await createPriceFeed(logger, web3, networker, getTime, config);
+
+    assert.isNull(expressionPriceFeed);
+  });
+
+  it("ExpressionPriceFeed: can find default price feeds", async function() {
+    const config = {
+      type: "expression",
+      lookback,
+      expression: "USDETH + ETH\\/BTC"
+    };
+
+    const expressionPriceFeed = await createPriceFeed(logger, web3, networker, getTime, config);
+
+    assert.isNotNull(expressionPriceFeed);
+  });
+
+  it("ExpressionPriceFeed: invalid config in customFeeds is ignored if unused", async function() {
+    const config = {
+      type: "expression",
+      expression: "2 + 5",
+      customFeeds: {
+        ETHBTC: {} // Invalid because it has no type.
+      }
+    };
+
+    const expressionPriceFeed = await createPriceFeed(logger, web3, networker, getTime, config);
+
+    assert.isNotNull(expressionPriceFeed);
+  });
+
+  it("ExpressionPriceFeed: constant expression", async function() {
+    const config = {
+      type: "expression",
+      expression: "1 + 2"
+    };
+
+    const expressionPriceFeed = await createPriceFeed(logger, web3, networker, getTime, config);
+
+    assert.isNotNull(expressionPriceFeed);
```

Comment:
nice test suite, wondering if its worth also testing for the error log?

Reply:
Done!